common/packets/event.h

```diff
@@ -25,11 +25,45 @@
 
 namespace RemoteCL
 {
+template<PacketType Type>
+struct EventCallback final : public Packet
+{
+	EventCallback() noexcept : Packet(Type) {}
+
+	// unique callback ID
+	uint32_t mID;
```

Comment:
Is there a reason uint32_t was chosen?
You named it "ID" so I'd expect this to be IDType.